### ISSUE TITLE
Using methods/properties of props triggers prop-types rule

### ISSUE THREAD

**Issue by mistadikay** [yannickcr/eslint-plugin-react#379]
When using methods or properties of any prop like `this.props.options.map(...)`, eslint throws a validation error `'options.map' is missing in props validation  react/prop-types`. The same goes for `.forEach`, `.filter`, etc. `options` propType looks like this:
```js
options: PropTypes.shape({
    text: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
    ]),
    value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
    ])
}),
```

My setup:
`babel-eslint@5.0.0-beta6`
`eslint@1.10.3`
`eslint-plugin-babel@3.0.0`
`eslint-plugin-react@3.13.1`

**Comment by mistadikay** [comment]
I'm sorry, it's me being dumb, `options` propType should be like this:
```js
options: PropTypes.arrayOf(
    PropTypes.shape({
        text: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number
        ]),
        value: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number
        ])
    })
),
```

In this case everything is fine. Closing
